### PR TITLE
[v4.2-rhel] Bump Buildah to v1.27.3

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Bump Buildah to v1.27.3 which itself bumps opencontainers/selinux to 1.10.2, has fixes for the sticky bit, allows containers to be restarted in a pod when they should be, a fix for a RHEL subscription issue,
 and s390x segv issues across RHEL 8.7 and 9.1 Zstreams

Addresses:
https://bugzilla.redhat.com/show_bug.cgi?id=2150416 https://bugzilla.redhat.com/show_bug.cgi?id=2150428 https://bugzilla.redhat.com/show_bug.cgi?id=2137294 https://bugzilla.redhat.com/show_bug.cgi?id=2149775 https://bugzilla.redhat.com/show_bug.cgi?id=2152027 https://bugzilla.redhat.com/show_bug.cgi?id=2152017 https://bugzilla.redhat.com/show_bug.cgi?id=2150433 https://bugzilla.redhat.com/show_bug.cgi?id=2150432 https://bugzilla.redhat.com/show_bug.cgi?id=2136928 https://bugzilla.redhat.com/show_bug.cgi?id=2149776 https://bugzilla.redhat.com/show_bug.cgi?id=2152026 https://bugzilla.redhat.com/show_bug.cgi?id=2152022 https://bugzilla.redhat.com/show_bug.cgi?id=2152042 https://bugzilla.redhat.com/show_bug.cgi?id=2152043

[NO NEW TESTS NEEDED]
Signed-off-by: tomsweeneyredhat <tsweeney@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -71,6 +71,9 @@ require (
 	gopkg.in/yaml.v3 v3.0.1
 )
 
-require github.com/docker/libnetwork v0.8.0-dev.2.0.20190625141545-5a177b73e316 // indirect
+require (
+	github.com/docker/libnetwork v0.8.0-dev.2.0.20190625141545-5a177b73e316 // indirect
+	github.com/varlink/go v0.4.0
```
Comment (mheon) [inline go.mod:76]: Where did this come from?

Comment (TomSweeneyRedHat) [reply go.mod:76]: I'm not sure and it looks like I've a dirty tree.  Let me try that again.

Comment (TomSweeneyRedHat) [reply go.mod]: Nice spot @mheon!  Something must have gone wonky with my first vendor.  Looking better now.

**Pull request merged**